src/sentry/static/sentry/app/views/settings/account/avatar.jsx

```diff
@@ -1,45 +1,55 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import styled from 'react-emotion';
 import createReactClass from 'create-react-class';
+import styled from 'react-emotion';
 
-import Panel from '../components/panel';
-import PanelBody from '../components/panelBody';
-import PanelHeader from '../components/panelHeader';
-import AlertActions from '../../../actions/alertActions';
+import {
+  addErrorMessage,
+  addSuccessMessage,
+} from '../../../actionCreators/settingsIndicator';
+import {t} from '../../../locale';
 import ApiMixin from '../../../mixins/apiMixin';
+import Avatar from '../../../components/avatar';
 import AvatarCropper from '../../../components/avatarCropper';
-import RadioGroup from '../components/forms/radioGroup';
 import LoadingError from '../../../components/loadingError';
 import LoadingIndicator from '../../../components/loadingIndicator';
-import {t} from '../../../locale';
+import Panel from '../components/panel';
+import PanelBody from '../components/panelBody';
+import PanelHeader from '../components/panelHeader';
+import RadioGroup from '../components/forms/radioGroup';
+import SentryTypes from '../../../proptypes';
 
-const AvatarSettings = createReactClass({
-  displayName: 'AvatarSettings',
+const AccountAvatar = createReactClass({
+  displayName: 'AccountAvatar',
 
   propTypes: {
     userId: PropTypes.number,
+    user: SentryTypes.User,
+    onSave: PropTypes.func,
   },
 
   mixins: [ApiMixin],
 
+  getDefaultProps() {
+    return {
+      onSave: () => {},
+    };
+  },
+
   getInitialState() {
     return {
-      user: null,
+      user: this.props.user,
       savedDataUrl: null,
       dataUrl: null,
       hasError: false,
     };
   },
 
-  componentDidMount() {
-    this.api.request(this.getEndpoint(), {
-      method: 'GET',
-      success: this.updateUserState,
-      error: () => {
-        this.setState({hasError: true});
-      },
-    });
+  componentWillReceiveProps(nextProps) {
+    // Update local state if defined in props
+    if (typeof nextProps.user !== 'undefined') {
```

Comment:
same tiny nit, about the undefined util, take or leave it

Reply:
I think `defined` here doesn't fit as much because `null` is a valid value (but actually we don't need this check at all, it was here because I thought this component was used in multiple places).